rollout/sync_test.go

```diff
@@ -639,3 +640,66 @@ func TestScaleDownDeploymentOnSuccess(t *testing.T) {
 
 	assert.NotNil(t, err)
 }
+
+// This tests validates that when there are old replicasets that have miss matched desired-count annotations aka they do
+// not match the spec.replicas field. When this happens we get stuck in a reconcile loop of only trying to scale replicasets
+// only because of a supposed scaling event. This test validates that we do not get stuck in this loop by checking that status.replicas
+// and status.readyreplicas are updated because those will not be updated in that loop.
+func TestIsScalingEventMissMatchedDesiredOldReplicas(t *testing.T) {
+	f := newFixture(t)
+	defer f.Close()
+
+	// these steps should be ignored
+	steps := []v1alpha1.CanaryStep{
+		{
+			SetWeight: int32Ptr(10),
+		},
+		{
+			Pause: &v1alpha1.RolloutPause{
+				Duration: v1alpha1.DurationFromInt(60),
+			},
+		},
+	}
+
+	r0 := newCanaryRollout("foo", 10, int32Ptr(4), steps, int32Ptr(0), intstr.FromInt(10), intstr.FromInt(0))
+	r0.Annotations[annotations.RevisionAnnotation] = "1"
+	rs0 := newReplicaSetWithStatus(r0, 3, 3)
+	rs0.Annotations[annotations.DesiredReplicasAnnotation] = "2"
+
+	r1 := bumpVersion(r0)
+
+	rs1 := newReplicaSetWithStatus(r1, 10, 10)
+	r1.Status.StableRS = rs1.Labels[v1alpha1.DefaultRolloutUniqueLabelKey]
+	r2 := bumpVersion(r1)
+	r2.Annotations[annotations.RevisionAnnotation] = "2"
+	//rs2 := newReplicaSetWithStatus(r2, , 0)
```

Comment:
Shouldn't this test contain a third RS (not new and not stable) with annotations that doesn't match the replica count so the old code fails? 

I don't see where is that RS in this test. 👀

Reply:
It will have 3 because it creates a new one during the reconcile.